fluss-server/src/test/java/com/alibaba/fluss/server/tablet/TabletServiceITCase.java

```diff
@@ -159,6 +164,42 @@ void testProduceLog() throws Exception {
                 .hasMessageContaining("Invalid required acks");
     }
 
+    @Test
+    void testProduceLogResponseReturnInOrder() throws Exception {
+        long tableId =
+                createTable(FLUSS_CLUSTER_EXTENSION, DATA1_TABLE_PATH, DATA1_TABLE_DESCRIPTOR);
```

Comment:
Use `DATA1_TABLE_DESCRIPTOR.withReplicationFactor(3)` to make the problem can be reproduced before fixing it.

Reply:
Done!